wdl/Module07XfBatchEffect.wdl

```diff
@@ -0,0 +1,679 @@
+            | cut -f2 \
+            | paste -s -d\, || true )
+    cut -f"$idxs" "~{prefix}.vcf2bed.bed" \
+    | sed 's/^name/\#VID/g' \
+    | gzip -c \
+    > "~{prefix}.frequencies.preclean.txt.gz"
+    #Clean frequencies
+    /opt/sv-pipeline/scripts/downstream_analysis_and_filtering/clean_frequencies_table.R \
+      "~{prefix}.frequencies.preclean.txt.gz" \
+      "~{prefix}.frequencies.txt"
+    ### Create table of freqs, irrespective of ancestry
+    #Cut to necessary columns
+    idxs=$( sed -n '1p' "~{prefix}.vcf2bed.bed" \
+            | sed 's/\t/\n/g' \
+            | awk -v OFS="\t" '{ if ($1=="name" || $1=="SVLEN" || $1=="SVTYPE" || $1=="AC" || $1=="AN") print NR }' \
+            | paste -s -d\, || true )
+    cut -f"$idxs" "~{prefix}.vcf2bed.bed" \
+    | sed 's/^name/\#VID/g' \
+    | gzip -c \
+    > "~{prefix}.frequencies.allPops.txt.gz"
+  >>>
+
+  output {
+    File freq_data = "~{prefix}.frequencies.txt.gz"
+    File freq_data_allPops = "~{prefix}.frequencies.allPops.txt.gz"
+  }
+
+  runtime {
+    cpu: select_first([runtime_attr.cpu_cores, default_attr.cpu_cores])
+    memory: select_first([runtime_attr.mem_gb, default_attr.mem_gb]) + " GiB"
+    disks: "local-disk " + select_first([runtime_attr.disk_gb, default_attr.disk_gb]) + " HDD"
+    bootDiskSizeGb: select_first([runtime_attr.boot_disk_gb, default_attr.boot_disk_gb])
+    docker: sv_pipeline_docker
+    preemptible: select_first([runtime_attr.preemptible_tries, default_attr.preemptible_tries])
+    maxRetries: select_first([runtime_attr.max_retries, default_attr.max_retries])
+  }
+}
+
+
+# Combine frequency data across batches
+task MergeFreqTables {
+  input{
+    Array[File] tables
+    File batches_list
+    String prefix
+    String sv_pipeline_docker
+    RuntimeAttr? runtime_attr_override
+  }
+  RuntimeAttr default_attr = object {
+    cpu_cores: 1, 
+    mem_gb: 16,
+    disk_gb: 100,
+    boot_disk_gb: 10,
+    preemptible_tries: 3,
+    max_retries: 1
+  }
+  RuntimeAttr runtime_attr = select_first([runtime_attr_override, default_attr])
+  command <<<
+    set -euo pipefail
+    #Get list of batch IDs and batch table paths
+    while read batch; do
+      echo "$batch"
+      find ./ -name "$batch.frequencies*txt.gz"
+    done < ~{batches_list} | paste - - \
+    > input.list
+    #Make sure all input files have the same number of lines
+    while read batch file; do
+      zcat "$file" | wc -l
+    done < input.list > nlines.list
+    nlines=$( sort nlines.list | uniq | wc -l )
+    if [ "$nlines" -gt 1 ]; then
+      echo "AT LEAST ONE INPUT FILE HAS A DIFFERENT NUMBER OF LINES"
+      exit 0
+    fi
+    #Prep files for paste joining
+    echo "PREPPING FILES FOR MERGING"
+    while read batch file; do
+        #Header
+        zcat "$file" | sed -n '1p' | cut -f1-3
+        #Body
+        zcat "$file" | sed '1d' \
+        | sort -Vk1,1 \
+        | cut -f1-3
+    done < <( sed -n '1p' input.list ) \
+    > header.txt
```

Comment:
This doesn't seem like just a header, it gets pasted to the processed files.. I guess I think of header as always being column headers, maybe calling this row_headers.txt would make this more clear.